R/mod_review_forms.R

```diff
@@ -241,46 +235,39 @@ mod_review_forms_server <- function(
           status      = if(input$form_reviewed) "old" else "new"
         ) 
       
-      golem::cat_dev("review row to add:\n")
-      golem::print_dev(review_row)
+      golem::cat_dev("review records to add:\n")
+      golem::print_dev(review_records)
       
       cat("write review progress to database\n")
       db_save_review(
-        review_row, 
+        review_records, 
         db_path = db_path,
-        # More tables can be added here if needed, to track process of 
-        # individual reviewers in individual tables:
-        tables = "all_review_data" 
+        table = "all_review_data" 
       )
       
-      # Contains multiple rows, one for each item.
       updated_rows_db <- db_get_review(
-        db_path, subject = review_row$subject_id, form = review_row$item_group
-        )[c(names(review_row), "event_name", "item_name", "form_repeat")]
-      # Within a form, only items with a changed review state are updated and 
-      # contain the new (current) time stamp. 
-      updated_rows_db <- updated_rows_db[
-        updated_rows_db$timestamp == review_row$timestamp[1], 
-        ]
+        db_path, id = review_records$id
+        ) |> 
+        dplyr::select(dplyr::all_of(names(review_records)))
       
-      review_row_db <- unique(updated_rows_db[names(review_row)])
-      if(identical(review_row_db, review_row)){
+      review_records_db <- updated_rows_db |> 
+        # Within a form, only items with a changed review state are updated and 
+        # contain the new (current) time stamp. 
+        dplyr::filter(timestamp == review_records$timestamp[1])
+      if(isTRUE(all.equal(review_records_db, review_records, check.attributes = FALSE))){
         cat("Update review data and status in app\n")
         r$review_data <- r$review_data |> 
-          dplyr::rows_update(
-            updated_rows_db, 
-            by = c("subject_id", "item_group", "event_name", "item_name", "form_repeat")
-          )
+          dplyr::rows_update(review_records, by = "id")
       }
       
-      updated_items_memory <- sort(with(r$review_data, item_name[
-        reviewer == review_row$reviewer[1] & timestamp == review_row$timestamp[1]
-      ]))
-      updated_items_db <- sort(updated_rows_db$item_name)
+      updated_items_memory <- review_records |> 
+        dplyr::left_join(r$review_data, by = "id", suffix = c("", ".y")) |> 
+        dplyr::select(dplyr::all_of(names(review_records))) |> 
```

Comment:
I am a bit confused with this. 
We want to check here that `r$review_data` contains the correct updated rows after the `rows_update` step above.

Why not just filter `r$review_data` on all the ids in `review_records`? Would that be a problem? 
```
r$review_data[r$review_data$id %in% review_records$id, ]
```

Furthermore, I think in the current setup, the review columns from r$review_data (status/comment/etc) are deleted and we are just checking the ones from `review_records` by using `suffix = c("", "y.")`. Thus, we are not checking what is actually stored in memory (in `r$review_data`).

Am I missing something here?

Also, minor: the name is now a bit confusing for me: before I was simply comparing a vector of items, hence the name `updated_items_memory`.

Reply:
Yes, I seem to have missed the plot when updating this code.

Reply:
I've added some small fixes (e.g. selected columns were not the same and therefore the comparison failed)